onnx/optimizer/passes/eliminate_unused_initializer.h

```diff
@@ -0,0 +1,80 @@
+// ATTENTION: The code in this file is highly EXPERIMENTAL.
+// Adventurous users should note that the APIs will probably change.
+
+#pragma once
+
+#include "onnx/optimizer/passes/optimize_pass.h"
+
+namespace ONNX_NAMESPACE { namespace optimization {
+
+struct EliminateUnusedInitializer final : public OptimizePass {
+  explicit EliminateUnusedInitializer()
+    : OptimizePass("eliminate_unused_initializer", API_TYPE::IR) {
+  }
+
+  void eliminate_unused_initializer(Graph& graph) {
+    std::vector<Tensor> new_initializers;
+    std::vector<std::string> new_initializer_names;
+    for (auto it = graph.begin(); it != graph.end(); ++it) {
+      auto* n = *it;
+      DescendOnGraphAttributes(n, [this](Graph& g){eliminate_unused_initializer(g);});
+      // put any initializer used as input of any node to new vector
+      for (auto* input : n->inputs()) {
```

Comment:
it will break if the same initializer is used twice in different nodes

Reply:
@dzhulgakov 
Thanks. 
By using `std::unordered_set`, it will not be a problem anymore.